Python/compile.c

```diff
@@ -3656,21 +3674,20 @@ compiler_try_star_except(struct compiler *c, stmt_ty s)
 
     compiler_use_next_block(c, reraise_star);
     ADDOP(c, PREP_RERAISE_STAR);
-    ADDOP(c, DUP_TOP);
-    ADDOP_JUMP(c, POP_JUMP_IF_TRUE, reraise);
+    ADDOP_JUMP(c, JUMP_IF_TRUE_OR_POP, reraise);
     NEXT_BLOCK(c);
 
-    /* Nothing to reraise - pop it */
-    ADDOP(c, POP_TOP);
-    ADDOP(c, POP_TOP);
+    /* Nothing to reraise */
```

Comment:
Is this safe? I know it's a bit weird, but it *is* possible to make a falsey `ExceptionGroup`:

```py
>>> class FalseyEG(ExceptionGroup):
...     def derive(self, excs):
...         return FalseyEG(self.message, excs)
...     def __bool__(self):
...         return False  # !!!
... 
>>> try:
...     raise FalseyEG("Bye!", [ValueError()])
... except *TypeError:
...     pass
... 
>>>
```

One way to avoid this would be to perform equivalent of an `is None` check here instead:

```c
    ADDOP(c, DUP_TOP);
    ADDOP_LOAD_CONST(c, Py_None);
    ADDOP(c, CMP_OP, 0);
    ADDOP_JUMP(c, POP_JUMP_IF_FALSE, reraise);
    NEXT_BLOCK(c);

    /* Nothing to reraise - pop it */
    ADDOP(c, POP_TOP);
```

Or, we could just have `PREP_RERAISE_STAR` perform the jump itself (it does have a free oparg, after all). Either solution seems fine to me.

Reply:
Good point!  I added a test for a falsy EG subclass.

I have made the requested changes; please review again.